Childrens-Social-Care-CPD/Controllers/CookieController.cs

```diff
@@ -0,0 +1,60 @@
+﻿using Childrens_Social_Care_CPD.Models;
+using Contentful.Core;
+using Contentful.Core.Models;
+using Contentful.Core.Search;
+using Microsoft.AspNetCore.Mvc;
+using System.Diagnostics;
+using Childrens_Social_Care_CPD.Enums;
+using Childrens_Social_Care_CPD.Constants;
+using Microsoft.AspNetCore.Diagnostics;
+using System.Security.Policy;
+using Microsoft.AspNetCore.Http.Extensions;
+using Microsoft.AspNetCore.WebUtilities;
+using Childrens_Social_Care_CPD.Interfaces;
+using Childrens_Social_Care_CPD.ActionFilters;
+
+namespace Childrens_Social_Care_CPD.Controllers
+{
+    [ServiceFilter(typeof(CPDActionFilter))]
+    public class CookieController : BaseController
+    {
+        private readonly ILogger<CookieController> _logger;
+        private readonly IContentfulDataService _contentfulDataService;
+
+        public CookieController(ILogger<CookieController> logger, IContentfulDataService contentfulDataService) 
+        {
+            _logger = logger;
+            _contentfulDataService = contentfulDataService;
+        }
+
+        /// <summary>
+        /// Method to set Google analytics cookie consent from user
+        /// </summary>
+        /// <param name="analyticsCookieConsent">
+        /// Cookie consent - Accept or Reject
+        /// </param>
+        /// <returns></returns>
+        [HttpGet]
+        public async Task<IActionResult> GetCookies(string analyticsCookieConsent, string pageName, string pageType, string referer, string sendingPageType, string sendingPage)
```

Comment:
could we think of another name for this action or even move this out from the controller?

Reply:
@rgparkins   We can rename it but we can not move it out as its following Post - Redirect - Get pattern for browser back button issue.